src/extension/postgres/extension.rs

```diff
@@ -0,0 +1,152 @@
+    }
+
+    /// Uses "WITH SCHEMA" on Create Extension Statement.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{*, tests_cfg::*, extension::postgres::ExtensionCreateStatement};
+    ///
+    /// let stmt = ExtensionCreateStatement::new("ltree")
+    ///     .schema("public")
+    ///     .to_owned();
+    ///
+    /// assert_eq!(
+    ///     stmt.to_string(),
+    ///     r#"CREATE EXTENSION ltree WITH SCHEMA public"#
+    /// );
+    /// ```
+    pub fn schema(&mut self, schema: &str) -> &mut Self {
+        self.schema = Some(schema.to_string());
+        self
+    }
+
+    /// Uses "VERSION" on Create Extension Statement.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{*, tests_cfg::*, extension::postgres::ExtensionCreateStatement};
+    ///
+    /// let stmt = ExtensionCreateStatement::new("ltree")
+    ///     .version("v0.1.0")
+    ///     .to_owned();
+    ///
+    /// assert_eq!(
+    ///     stmt.to_string(),
+    ///     r#"CREATE EXTENSION ltree VERSION v0.1.0"#
+    /// );
+    /// ```
+    pub fn version(&mut self, version: &str) -> &mut Self {
+        self.version = Some(version.to_string());
+        self
+    }
+
+    /// Uses "IF NOT EXISTS" on Create Extension Statement.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{*, tests_cfg::*, extension::postgres::ExtensionCreateStatement};
+    ///
+    /// let stmt = ExtensionCreateStatement::new("ltree")
+    ///     .if_not_exists()
+    ///     .to_owned();
+    ///
+    /// assert_eq!(
+    ///     stmt.to_string(),
+    ///     r#"CREATE EXTENSION IF NOT EXISTS ltree"#
+    /// );
+    /// ```
+    pub fn if_not_exists(&mut self) -> &mut Self {
+        self.if_not_exists = true;
+        self
+    }
+
+    /// Uses "CASCADE" on Create Extension Statement.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{*, tests_cfg::*, extension::postgres::ExtensionCreateStatement};
+    ///
+    /// let stmt = ExtensionCreateStatement::new("ltree")
+    ///     .cascade()
+    ///     .to_owned();
+    ///
+    /// assert_eq!(
+    ///     stmt.to_string(),
+    ///     r#"CREATE EXTENSION ltree CASCADE"#
+    /// );
+    /// ```
+    pub fn cascade(&mut self) -> &mut Self {
+        self.cascade = true;
+        self
+    }
+}
+
+impl ToString for ExtensionCreateStatement {
+    fn to_string(&self) -> String {
+        let mut stmt = String::from("CREATE EXTENSION");
+
+        if self.if_not_exists {
+            stmt.push_str(" IF NOT EXISTS");
+        }
+
+        stmt.push_str(&format!(" {}", self.name));
+
+        if let Some(schema) = self.schema.as_ref() {
+            stmt.push_str(&format!(" WITH SCHEMA {}", schema));
+        }
+
+        if let Some(version) = self.version.as_ref() {
+            stmt.push_str(&format!(" VERSION {}", version));
+        }
+
+        if self.cascade {
+            stmt.push_str(" CASCADE");
+        }
+
+        stmt
+    }
+}
```

Comment:
@billy1624 I'm not 100% sure on this bit, wdyt? Given that this is only supported by PSQL I thought theres no need for specifying database backend 🤔 as we do for Query for instance.

Reply:
Hey @EstebanBorai, thanks for the quick PR! There is more conventional way of doing it :)
Check below for how we write the Postgres type statement.

- https://github.com/SeaQL/sea-query/blob/b0d1a4083a79b2872d9fd9ced0bdff4c310880a0/src/extension/postgres/types.rs#L102-L132
- https://github.com/SeaQL/sea-query/blob/b0d1a4083a79b2872d9fd9ced0bdff4c310880a0/src/backend/postgres/types.rs#L4-L117